packages/aws-client/tests/lib/test-S3MultipartUploads.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+const test = require('ava');
+const {
+  buildCompleteMultipartUploadParams,
+  buildUploadPartCopyParams,
+  createMultipartChunks
+} = require('../../lib/S3MultipartUploads');
+
+test('createMultipartChunks() returns the correct chunks', (t) => {
+  t.deepEqual(
+    createMultipartChunks(0),
+    []
+  );
+
+  t.deepEqual(
+    createMultipartChunks(9, 10),
+    [
+      { start: 0, end: 8 }
+    ]
+  );
+
+  t.deepEqual(
+    createMultipartChunks(10, 10),
+    [
+      { start: 0, end: 9 }
+    ]
+  );
+
+  t.deepEqual(
+    createMultipartChunks(11, 10),
+    [
+      { start: 0, end: 9 },
+      { start: 10, end: 10 }
+    ]
+  );
+
+  t.deepEqual(
+    createMultipartChunks(12, 10),
+    [
+      { start: 0, end: 9 },
+      { start: 10, end: 11 }
+    ]
+  );
+});
```

Comment:
I suggest splitting each of these `deepEqual` cases out into separate tests, as a general practice, as these represent parameterized tests and you want them _all_ to be executed, regardless of the failure of the others. In other words, if the first `deepEqual` fails, all of the remaining ones are skipped, and you won't see _all_ of the failures at once (if there would be multiple).

However, given this is pretty trivial, I'd be okay if you would prefer to leave things as they are, but want to at least plant the general idea in your mind.

Reply:
👍